keycloak.yaml

```diff
@@ -54,6 +54,15 @@ pipeline:
         ln -sf /usr/share/java/keycloak/bin/$i ${{targets.destdir}}/usr/bin/$i
       done
 
+# images like keycloak-operator may need it as they expect it to find keycloak in /opt/keycloak
+# For ref: https://github.com/keycloak/keycloak/blob/4b194d00bed51458acb3d125eba9a0ba654c930a/operator/Dockerfile#L11
+subpackages:
+  - name: keycloak-compat
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.subpkgdir}}"/opt
+          cp -r ${{targets.destdir}}/usr/share/java/keycloak "${{targets.subpkgdir}}"/opt/
```

Comment:
should this be a symlink rather than copy of the binary?

Reply:
I thought of that first but the problem that I thought it might bring is that its not a single file that I want to symlink but probably different directories and symlink sometimes afaik creates problems with it? This is how it looks like:

curl -sL [https://packages.wolfi.dev/os/aarch64](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/APKINDEX.tar.gz)/[keycloak-24.0.3-r2.apk](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377) | [tar -tvz](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java/keycloak/?all=true) [usr/share/java/](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java)keycloak


-rw-r--r-- 0/0           11358 2024-05-01 11:59 [LICENSE.txt](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java/keycloak/LICENSE.txt)
-rw-r--r-- 0/0             492 2024-05-01 11:59 [README.md](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java/keycloak/README.md)
drwxr-xr-x 0/0               0 2024-05-01 11:59 [bin](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java/keycloak/bin/)
drwxr-xr-x 0/0               0 2024-05-01 11:59 [conf](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java/keycloak/conf/)
drwxr-xr-x 0/0               0 2024-05-01 11:59 [lib](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java/keycloak/lib/)
drwxr-xr-x 0/0               0 2024-05-01 11:59 [providers](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java/keycloak/providers/)
drwxr-xr-x 0/0               0 2024-05-01 11:59 [themes](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java/keycloak/themes/)
-rw-r--r-- 0/0              26 2024-05-01 11:59 [version.txt](https://apk.dag.dev/https/packages.wolfi.dev/os/aarch64/keycloak-24.0.3-r2.apk@sha1:7052225b19f2497df56bd383a6179dc7cf197377/usr/share/java/keycloak/version.txt)

Reply:
Another example of similar scenario probably https://github.com/wolfi-dev/os/commit/f67a88a2d88ebe410becbb42e5df03727b0a9331

Reply:
Yeah, I think symlinks are preferable else installing the main package and the subpackage will result in two keycloak binaries installed.

Reply:
Cool, thanks @rawlingsj! Done and tested locally, it is working!